runtime/lib/capi/RuntimeCAPI.cpp

```diff
@@ -229,14 +229,22 @@ QirArray *__quantum__rt__qubit_allocate_array(int64_t num_qubits)
     RT_ASSERT(Catalyst::Runtime::CTX->getMemoryManager() != nullptr);
     RT_ASSERT(num_qubits >= 0);
 
-    QirArray *qubit_array = __quantum__rt__array_create_1d(sizeof(QubitIdType), num_qubits);
-    const auto &&qubit_vector =
+    // For first prototype, we just want to make this work.
+    // But ideally, I think the device should determine the representation.
+    // Essentially just forward this to the device library.
+    // And the device library can choose how to handle everything.
+    //
+    // Because QirArray* is opaque, we can return anything.
+    // But because this is a C function, we should return a C-like item.
+    // So, let's just return a pointer.
```

Comment:
I will say returning a pointer is the one thing we _must_ do (to be consistent with the QIR types). `QirArray` is the opaque part, but the pointer in `QirArray*` is the concrete part.

Reply:
What I meant in this comment was to not return a smart pointer because this is a C interface facing function.

Reply:
I'll change the comment. I admit that it is not clear. Thanks!